### PR TITLE
test(`e2e`): improve ZEVM Message Passing E2E tests and reformat E2E tests structure

### PR DESCRIPTION
**Pull request by lumtis**
# Description

- Add assertion that check ZETA are correctly transferred or sent back to the Message Passing contract when reverting
- Add new tests: revert failure evm->zevm and zevm->evm
- Some other reorganization
  - Reorder E2E test names per types
  - Create a file for each E2E tests

### CODE REVIEW

File: e2e/e2etests/test_bitcoin_withdraw_segwit.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package e2etests
+
+import (
+	"github.com/btcsuite/btcutil"
+	"github.com/zeta-chain/zetacore/e2e/runner"
+)
+
+func TestBitcoinWithdrawSegWit(r *runner.E2ERunner, args []string) {
```
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/e2etests/test_bitcoin_withdraw_segwit.go:8]: >do we want to have convention to have 1 test per file?

Yes, this is the idea, it was sometimes annoying finding a specific test

File: e2e/e2etests/helper_bitcoin.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
+package e2etests
```
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/e2etests/helper_bitcoin.go:1]: This is just to highlight that this is not related to a test, also allow to group the helper files all together

File: e2e/e2etests/e2etests.go [hunk 3]
```diff
@@ -4,234 +4,304 @@ import (
...
+	TestContextUpgradeName = "context_upgrade"
+	TestMyTestName         = "my_test"
+	TestDonationEtherName  = "donation_ether"
 
+	/*
+	 Stress tests
+	 Test stressing networks with many cross-chain transactions
+	*/
 	TestStressEtherWithdrawName = "stress_eth_withdraw"
 	TestStressBTCWithdrawName   = "stress_btc_withdraw"
```
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/e2etests/e2etests.go:89]: I think there were some issues running these tests, I will have to have a new look at it

File: e2e/e2etests/e2etests.go [hunk 4]
```diff
@@ -4,234 +4,304 @@ import (
 	"github.com/zeta-chain/zetacore/e2e/runner"
 )
 
-// TODO : Add smoke test for abort refund
+// List of all e2e test names to be used in zetae2e
+// TODO : E2E smoke test for abort refund
```
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/e2etests/e2etests.go:8]: The issue is mentioned just below and has been completed, we can actually remove this line

File: e2e/e2etests/e2etests.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/e2etests/e2etests.go:89]: https://github.com/zeta-chain/node/issues/2194

File: e2e/e2etests/test_bitcoin_withdraw_segwit.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/e2etests/test_bitcoin_withdraw_segwit.go:8]: 9276a6d8

File: e2e/e2etests/test_message_passing_zevm_to_evm.go [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+	// Check ZETA balance on EVM TestDApp and check new balance between previous balance and previous balance + amount
+	// Contract receive less than the amount because of the gas fee to pay
+	newBalanceEVM, err := r.ZetaEth.BalanceOf(&bind.CallOpts{}, r.EvmTestDAppAddr)
+	if err != nil {
+		panic(err)
+	}
+	previousBalanceAndAmountEVM := big.NewInt(0).Add(previousBalanceEVM, amount)
+
+	// check higher than previous balance and lower than previous balance + amount
+	if newBalanceEVM.Cmp(previousBalanceEVM) <= 0 || newBalanceEVM.Cmp(previousBalanceAndAmountEVM) > 0 {
```
Comment (ws4charlie) [inline e2e/e2etests/test_message_passing_zevm_to_evm.go:128]: `newBalanceEVM.Cmp(previousBalanceEVM) <= 0 || newBalanceEVM.Cmp(previousBalanceAndAmountEVM) >= 0`
shouldn't be this?

File: e2e/e2etests/test_message_passing_zevm_to_evm_revert.go [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+	// New balance is increased because ZETA are sent from the sender but sent back to the contract
+	// Contract receive less than the amount because of the gas fee to pay
+	newBalanceZEVM, err := r.WZeta.BalanceOf(&bind.CallOpts{}, r.ZevmTestDAppAddr)
+	if err != nil {
+		panic(err)
+	}
+	previousBalanceAndAmountZEVM := big.NewInt(0).Add(previousBalanceZEVM, amount)
+
+	// check higher than previous balance and lower than previous balance + amount
+	if newBalanceZEVM.Cmp(previousBalanceZEVM) <= 0 || newBalanceZEVM.Cmp(previousBalanceAndAmountZEVM) > 0 {
```
Comment (ws4charlie) [inline e2e/e2etests/test_message_passing_zevm_to_evm_revert.go:124]: same `newBalanceZEVM.Cmp(previousBalanceAndAmountZEVM) >= 0` ?

**Review by ws4charlie — COMMENTED** [review]
looks good, just two comments (not 100% sure)

**Review by ws4charlie — APPROVED** [review]

**Pull request merged**